frontend/src/components/nodes/ValidatorRow.tsx

```diff
@@ -44,32 +45,91 @@ const WarningText = styled("td", {
 });
 
 interface Props {
-  node: ValidationNodeInfo;
+  validator: ValidatorFullData;
   index: number;
   totalStake: BN;
   cumulativeStake: BN;
   isNetworkHolder: boolean;
+  seatPrice?: string;
 }
 
 const ZERO = new BN(0);
 const EXTRA_PRECISION_MULTIPLIER = Math.pow(10, 2 + FRACTION_DIGITS);
 
+const getStakingStatus = (
+  validator: ValidatorFullData,
+  seatPrice?: string
+): StakingStatus | null => {
+  if (validator.currentEpoch) {
+    if (validator.nextEpoch) {
+      return "active";
+    } else {
+      return "leaving";
+    }
+  } else {
+    if (validator.nextEpoch) {
+      return "joining";
+    } else {
+      if (validator.afterNextEpoch) {
+        return "proposal";
+      } else {
+        if (!seatPrice) {
+          return null;
+        }
+        const contractStake = validator.contractStake
+          ? new BN(validator.contractStake)
+          : undefined;
+        if (!contractStake) {
+          return null;
+        }
+        const seatPriceBN = new BN(seatPrice);
+        if (contractStake.gte(seatPriceBN)) {
+          return "onHold";
+        } else if (contractStake.gte(seatPriceBN.muln(20).divn(100))) {
+          return "newcomer";
+        } else {
+          return "idle";
+        }
+      }
+    }
+  }
+};
+
 const ValidatorRow: React.FC<Props> = React.memo(
-  ({ node, index, totalStake, cumulativeStake, isNetworkHolder }) => {
+  ({
+    validator,
+    index,
+    totalStake,
+    cumulativeStake,
+    isNetworkHolder,
+    seatPrice,
+  }) => {
     const { t } = useTranslation();
     const [isRowActive, setRowActive] = React.useState(false);
     const switchRowActive = React.useCallback(() => setRowActive((x) => !x), [
       setRowActive,
     ]);
 
+    const visibleStake =
+      validator.currentEpoch?.stake ??
+      validator.nextEpoch?.stake ??
+      validator.afterNextEpoch?.stake ??
+      validator.contractStake;
+
+    const currentStake = validator.currentEpoch?.stake;
+    const nextVisibleStake =
+      validator.nextEpoch?.stake ?? validator.afterNextEpoch?.stake;
```

Comment:
So we have a few cases in here, with three stakes that may exist or not: `currentEpoch.stake`, `nextEpoch.stake`, `afterNextEpoch.stake`. We can make a table of what kind of deltas we want to show.

In case proposal exists:
|       | current                 | !current         |
|-------|-------------------------|------------------|
| next  | current -> next         | next -> proposal |
| !next | current -> proposal     | 0 -> proposal    |

In case proposal does not exist:
|       | current                 | !current         |
|-------|-------------------------|------------------|
| next  | current -> next         | next -> proposal |
| !next | current -> 0            | -                |

Is that what you expect or not?

Reply:
These tables look too complex to me, and it seems they are not what I wanted to say.

1. Well, my fix was incorrect since I did not rename `nextVisibleStake` variable name. Your logic was correct regarding this naming, but I actually wanted us to display not the `nextVisibleStake` diff, but rather `mostRecentStake` diff
2. `mostRecentStake` is `validator.afterNextEpoch?.stake ?? validator.nextEpoch?.stake`

Reply:
> mostRecentStake is validator.afterNextEpoch?.stake ?? validator.nextEpoch?.stake

But in case of an active validator it will be the delta between the current epoch and the proposal, is that actually what we want to do?

Reply:
@luixo yes, this is what people expect to see, I believe. In the future we might want to educate people better about the fact that the next epoch validators list is already frozen, and the proposals are landing to the epoch that is the one that is next after the next one, but for now I feel it is fine to just hide that unnecessary technical detail from users.